src/Azure/Azure.Quantum.Client/Authentication/DefaultQuantumCredential.cs

```diff
@@ -55,7 +55,11 @@ public virtual IEnumerable<TokenCredential> Sources
             {
                 yield return CredentialFactory.CreateCredential(CredentialType.TokenFile);
                 yield return CredentialFactory.CreateCredential(CredentialType.Environment, SubscriptionId);
-                yield return CredentialFactory.CreateCredential(CredentialType.ManagedIdentity, SubscriptionId);
+                // Create a ManagedIdentityCredential with a 1-second timeout to work around delays in non-Azure environments.
+                // In non-Azure environments, ManagedIdentityCredential can cause long delays before failing to resolve.
+                // In Azure environments, in practice, 1 second seems to be enough time for the managed identity to be resolved.
+                // (See: https://github.com/Azure/azure-sdk-for-net/issues/24767)
+                yield return new ManagedIdentityCredential(options: new TokenCredentialOptions { Retry = { NetworkTimeout = TimeSpan.FromSeconds(1), MaxRetries = 0 } });
```

Comment:
1 second does seem pretty aggressive. Does it expect to receive network replies or resolve loading any artifacts (other assemblies, config settings, etc.) within that time to be successful? If so, maybe something like 5 or 10 seconds might be a little more conservative, but still well within the 2 min timeout we're seeing.

(And I'm speculating, so if that's totally wrong and things will fail instantly if they're going to fail, then this is fine).

Reply:
I should have read the opening comment on the PR before my review, not after :-)  Looks good to go then. Thanks!